python/hail/table.py

```diff
@@ -409,8 +431,12 @@ def is_copy(ast, name):
             preserved_key = preserved_key_new = new_key = None
             row = value_struct if value_struct is not None else hl.struct()
 
-        base, cleanup = self._process_joins(row)
-        analyze(caller, row, self._row_indices)
+        if reprocess_joins:
+            base, cleanup = self._process_joins(row)
+            analyze(caller, row, self._row_indices)
+        else:
+            # ensure analyze and process_joins were done beforehand
```

Comment:
Does this need to be done?

Reply:
I added this comment because all prior uses of `_select` rely on the expressions to be analyzed and joins processed inside the `_select` method. `_select` with `reprocess_joins=False` will fail if there are joins in the expressions. Are you suggesting I remove the comment?

Do you think `do_process_joins` is a better parameter name? We aren't actually reprocessing in the `True` case, just processing for the first time, and I don't want to shadow `process_joins`.

Reply:
Actually, I think I can pull out the `analyze` from the conditional now, will try.

Reply:
Nevermind, re-analyzing after process joins throws the usual source error.